citesphere/src/main/java/edu/asu/diging/citesphere/web/user/AsyncUpdateCitationsProcessor.java

```diff
@@ -0,0 +1,87 @@
+package edu.asu.diging.citesphere.web.user;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.social.zotero.api.ZoteroUpdateItemsStatuses;
+import org.springframework.social.zotero.exception.ZoteroConnectionException;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.asu.diging.citesphere.core.exceptions.CitationIsOutdatedException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.model.bib.ICitation;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Component
+public class AsyncUpdateCitationsProcessor {
+    @Autowired
+    private ICitationManager citationManager;
+
+    private final Map<String, Future<ZoteroUpdateItemsStatuses>> taskTracker;
+
+    public AsyncUpdateCitationsProcessor() {
+        this.taskTracker = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * This method updates citations.
+     * 
+     * @param user          User accessing Zotero
+     * @param zoteroGroupId GroupId of the citations
+     * @param citations     citations that have to be updated
+     * @return returns AsyncUpdateCitationsResponse object that contains task id,
+     *         response and status.
+     */
+    public AsyncUpdateCitationsResponse updateCitations(IUser user, String zoteroGroupId, List<ICitation> citations)
+            throws JsonProcessingException, ZoteroConnectionException, CitationIsOutdatedException,
+            ZoteroHttpStatusException, InterruptedException, ExecutionException {
+        String taskID = UUID.randomUUID().toString();
+        Future<ZoteroUpdateItemsStatuses> futureTask = citationManager.updateCitations(user, zoteroGroupId, citations);
```

Comment:
I would have another class maybe an asyncCitationManager have an async method that calls the zotero manager etc.